src/backend/distributed/transaction/transaction_management.c

```diff
@@ -362,6 +370,17 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 
 			ResetGlobalVariables();
 
+			/*
+			 * Clear MetadataCache table if we're aborting from a CREATE EXTENSION Citus
+			 * so that any created OIDs from the table are cleared and invalidated. We
+			 * also set CreateCitusTransactionLevel to 0 since that process has been aborted
+			 */
+			if (GetCitusCreationLevel() == GetCurrentTransactionNestLevel())
```

Comment:
Maybe with an `Assert(GetCitusCreationLevel() == 1)` or something?

Reply:
I think checking that `GetCitusCreationLevel == 1` is more clear to indicate that the level should be down to 1 now since we are committing/aborting the original transaction